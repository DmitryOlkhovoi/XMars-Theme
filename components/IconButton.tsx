import React, { FunctionComponent, ButtonHTMLAttributes } from 'react';
import Button, { ButtonProps } from './Button';
import Icon from './Icon';

export interface Props extends ButtonProps, ButtonHTMLAttributes<HTMLButtonElement> {
    name: string,
}

const IconButton: FunctionComponent<Props> = (props) => {
    return (
       <Button {...props}>
           <Icon {...props}></Icon>
       </Button>
    );
}

export default IconButton;