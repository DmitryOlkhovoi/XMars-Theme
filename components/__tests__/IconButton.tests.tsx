import React from 'react';
import IconButton from '../IconButton';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('IconButton component', () => {
  test('Default render', () => {
    const component = renderer.create(
      <IconButton name="more" />,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Primary render', () => {
    const component = renderer.create(
      <IconButton primary name="more" />,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Circle render', () => {
    const component = renderer.create(
      <IconButton circle name="more" />,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Ripple render', () => {
    const component = renderer.create(
      <IconButton ripple name="more" />,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('additionalClass render', () => {
    const component = renderer.create(
      <IconButton additionalClass="foo bar" name="more" />,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('onClick', () => {
    const onClick = jest.fn();

    const component = shallow(
      <IconButton onClick={onClick} name="more" />,
    );

    component.simulate('click');
  
    expect(onClick).toHaveBeenCalled();
  });
});