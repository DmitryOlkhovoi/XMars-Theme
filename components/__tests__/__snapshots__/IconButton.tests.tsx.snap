// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component Circle render 1`] = `
<button
  circle={true}
  className="btn circle"
  name="more"
  onClick={[Function]}
>
  <i
    circle={true}
    className="icon-more undefined"
  />
</button>
`;

exports[`IconButton component Default render 1`] = `
<button
  className="btn"
  name="more"
  onClick={[Function]}
>
  <i
    className="icon-more undefined"
  />
</button>
`;

exports[`IconButton component Primary render 1`] = `
<button
  className="btn primary"
  name="more"
  onClick={[Function]}
>
  <i
    className="icon-more undefined"
    primary={true}
  />
</button>
`;

exports[`IconButton component Ripple render 1`] = `
<button
  className="btn with-ripple"
  name="more"
  onClick={[Function]}
  ripple={true}
>
  <i
    className="icon-more undefined"
    ripple={true}
  />
</button>
`;

exports[`IconButton component additionalClass render 1`] = `
<button
  className="btn foo bar"
  name="more"
  onClick={[Function]}
>
  <i
    className="icon-more foo bar"
  />
</button>
`;
